http-clients/apache-client/src/main/java/software/amazon/awssdk/http/apache/internal/impl/ApacheHttpRequestFactory.java

```diff
@@ -150,7 +153,8 @@ private HttpRequestBase wrapEntity(HttpExecuteRequest request,
          */
         if (request.contentStreamProvider().isPresent()) {
             HttpEntity entity = new RepeatableInputStreamRequestEntity(request);
-            if (!request.httpRequest().firstMatchingHeader(HttpHeaders.CONTENT_LENGTH).isPresent()) {
+            boolean notChunked = !request.httpRequest().matchingHeaders(TRANSFER_ENCODING).contains(CHUNKED);
```

Comment:
To avoid checking the header again, we can probably just check `entity.isChunked` here

Reply:
right, updated